src/main/scala/edu/twitter/classification/Classifier.scala

```diff
@@ -0,0 +1,51 @@
+package edu.twitter.classification
+
+import edu.twitter.model.{GradientBoostingModel, SentimentModelDataCreator, TweetsLoader}
+import edu.twitter.streaming.TwitterStream
+import org.apache.spark.mllib.feature.HashingTF
+import org.apache.spark.streaming.StreamingContext
+import org.apache.spark.streaming.dstream.DStream
+
+/** Representation of the classified tweet, we may add
+  * more fields to it later. */
+case class ClassifiedTweet(label: Double, tweetText: String)
+
+/** Responsible for building `TweeterStream` and the `Classification Model`
+  * and classifying the stream with the model.
+  *
+  * @param ssc StreamingContext, used for model and
+  *            stream creation
+  */
+class Classifier(ssc: StreamingContext) {
+
+  /**
+    * Build the `Classification Model` and a `TweeterStream`
+    * and return a stream of `ClassifiedTweets`.
+    *
+    * @return stream of `ClassifiedTweets`
+    */
+  def createClassifiedStream(): DStream[ClassifiedTweet] = {
+    val tweets = new TwitterStream(ssc).createStream()
+    val model = createModel()
+    val hashingTF = new HashingTF(2000)
+    val supportedLangIso = Set("en", "eng")
+    val classifiedStream = for {
```

Comment:
Is this equal to reduce or collect in the stream api?

Reply:
No, for comprehensions in Scala is desugared to `flatmap` and `map`, these are all transformations to the original data not actions like the `collect`. the difference is that actions force the computation to happen, unlike transformations.